Add ventilation flux to urban canyon

olyson opened the pull request:
### Description of changes
The effects of ventilation (exchange of building air with canopy air) are accounted for in the energy budget inside the building.  The effects on urban canopy air are not.  This remedies that.

### Specific notes

Contributors other than yourself, if any:  Originally identified by Lingying Wang and Dan Li.

CTSM Issues Fixed (include github issue:
  Fixes #1526 -- Switch FATES methane test for methane off
  Fixes #509 --- irrigate in 1850 is off for runs with use_crop but on for those without
  Fixes #1578 -- Need to replace SSP370 ndep file with new one
  Fixes #1631 -- ndep file for SSP2-4.5 is incorrect

Are answers expected to change (and if so in what way)?  Urban-related fields and gridcell average fields will change for configurations that use the CLM5 building energy model.  CLM45 configurations (which use the CLM45 building energy model) should not.

Any User Interface Changes (namelist or namelist defaults changes)?  No.

Testing performed, if any:
Ran clm_short test suite on cheyenne.  All pass except for baseline comparisons as expected.  Ran one CLM45 test (ERS_D_Ld6.f10_f10_mg37.I1850Clm45BgcCrop.cheyenne_intel.clm-clm50CMIP6frc) which was BFB as expected.

olyson commented:
Standard diagnostics that demonstrate very small changes in gridcell averages of important variables are here:

https://webext.cgd.ucar.edu/I2000/ctsm51_ctsm51d61_1deg_GSWP3V1_CON_VENT_2000/lnd/ctsm51_ctsm51d61_1deg_GSWP3V1_CON_VENT_2000.1991_2010-ctsm51_ctsm51d61_1deg_GSWP3V1_CON_2000.1991_2010/set2/set2.html

Effects on urban related variables are documented here:

[CTSM51d61_BUILDENERGY_FVENT_Analysis_Pub.xlsx](https://github.com/ESCOMP/CTSM/files/7444845/CTSM51d61_BUILDENERGY_FVENT_Analysis_Pub.xlsx)

I've repeated testing on my latest commit and think this PR is ready for review.  I've changed it from draft status.

ekluzek commented:
Just to find out what fields change answers I ran the test: /SMS_Lm1.f19_g17.I1850Clm50BgcCropCmip6waccm.cheyenne_intel.clm-basic and see the following things that change answers:

> (npl) ctsm_worktrees/answer_changes> grep RMS /glade/scratch/erik/SMS_Lm1.f19_g17.I1850Clm50BgcCropCmip6waccm.cheyenne_intel.clm-basic.C.20220103_160716_wx0097/run/SMS_Lm1.f19_g17.I1850Clm50BgcCropCmip6waccm.cheyenne_intel.clm-basic.C.20220103_160716_wx0097.cpl.hi.0001-02-01-00000.nc.cprnc.out
>  RMS lndImp_Fall_evap                 9.2419E-09            NORMALIZED  2.2011E-03
>  RMS lndImp_Fall_lat                  2.3300E-02            NORMALIZED  2.1912E-03
>  RMS lndImp_Fall_lwup                 1.0602E-02            NORMALIZED  3.6556E-05
>  RMS lndImp_Fall_sen                  6.0793E-02            NORMALIZED  2.9825E-03
>  RMS lndImp_Fall_swnet                6.4721E-04            NORMALIZED  1.8779E-05
>  RMS lndImp_Fall_taux                 9.5907E-05            NORMALIZED  1.5160E-03
>  RMS lndImp_Fall_tauy                 9.5907E-05            NORMALIZED  1.5160E-03
>  RMS lndImp_Flrl_rofgwl               8.2994E-23            NORMALIZED  5.5690E-17
>  RMS lndImp_Flrl_rofsub               1.1300E-08            NORMALIZED  2.4604E-03
>  RMS lndImp_Flrl_rofsur               2.0606E-08            NORMALIZED  1.0559E-02
>  RMS lndImp_Sl_anidf                  1.1660E-05            NORMALIZED  1.4563E-05
>  RMS lndImp_Sl_anidr                  5.1951E-06            NORMALIZED  6.5884E-06
>  RMS lndImp_Sl_avsdf                  1.4472E-05            NORMALIZED  1.6646E-05
>  RMS lndImp_Sl_avsdr                  6.6529E-06            NORMALIZED  7.6299E-06
>  RMS lndImp_Sl_fv                     1.5642E-04            NORMALIZED  7.5181E-04
>  RMS lndImp_Sl_qref                   3.4671E-07            NORMALIZED  8.3580E-05
>  RMS lndImp_Sl_ram1                   5.4600E-02            NORMALIZED  3.0138E-04
>  RMS lndImp_Sl_snowh                  6.9988E-06            NORMALIZED  5.4118E-06
>  RMS lndImp_Sl_t                      2.2908E-03            NORMALIZED  8.6624E-06
>  RMS lndImp_Sl_tref                   1.5436E-03            NORMALIZED  5.8162E-06
>  RMS lndImp_Sl_u10                    3.0708E-04            NORMALIZED  2.3750E-04
>  RMS lndExp_Flrr_volr                 3.0848E-06            NORMALIZED  1.0990E-03
>  RMS lndExp_Flrr_volrmch              1.9974E-06            NORMALIZED  8.3471E-04
>  RMS rofImp_Flrr_volr                 4.6191E-06            NORMALIZED  1.6254E-03
>  RMS rofImp_Flrr_volrmch              3.8011E-06            NORMALIZED  1.5682E-03
>  RMS rofImp_Forr_rofl                 1.5384E-08            NORMALIZED  9.1660E-03
>  RMS rofExp_Flrl_rofgwl               3.7251E-23            NORMALIZED  1.3761E-16
>  RMS rofExp_Flrl_rofsub               6.5188E-09            NORMALIZED  4.7852E-03
>  RMS rofExp_Flrl_rofsur               6.9829E-09            NORMALIZED  1.1681E-02
> (npl) ctsm_worktrees/answer_changes> grep RMS /glade/scratch/erik/SMS_Lm1.f19_g17.I1850Clm50BgcCropCmip6waccm.cheyenne_intel.clm-basic.C.20220103_160716_wx0097/run/SMS_Lm1.f19_g17.I1850Clm50BgcCropCmip6waccm.cheyenne_intel.clm-basic.C.20220103_160716_wx0097.mosart.h0.0001-01.nc.cprnc.out
>  RMS DIRECT_DISCHARGE_TO_OCEAN_LIQ    4.4519E-04            NORMALIZED  1.1377E-03
>  RMS RIVER_DISCHARGE_OVER_LAND_LIQ    7.5318E-02            NORMALIZED  4.7350E-04
>  RMS TOTAL_DISCHARGE_TO_OCEAN_LIQ     1.2112E-02            NORMALIZED  4.0162E-03
> (npl) ctsm_worktrees/answer_changes> grep RMS /glade/scratch/erik/SMS_Lm1.f19_g17.I1850Clm50BgcCropCmip6waccm.cheyenne_intel.clm-basic.C.20220103_160716_wx0097/run/SMS_Lm1.f19_g17.I1850Clm50BgcCropCmip6waccm.cheyenne_intel.clm-basic.C.20220103_160716_wx0097.clm2.h1.0001-01.nc.cprnc.out
>  RMS FCTR                             1.9901E+00            NORMALIZED  1.9401E-01
>  RMS FGEV                             2.8150E+00            NORMALIZED  3.1399E-01
>  RMS FIRA                             2.9903E+00            NORMALIZED  5.6355E-02
>  RMS FIRE                             7.4831E-01            NORMALIZED  2.5841E-03
>  RMS FSA                              2.4764E-01            NORMALIZED  2.8328E-03
>  RMS FSH                              9.9280E+00            NORMALIZED  3.4737E-01
>  RMS FSR                              1.5417E-01            NORMALIZED  3.3372E-03
>  RMS TREFMNAV                         7.5603E-02            NORMALIZED  2.7979E-04
>  RMS TREFMXAV                         5.6518E-02            NORMALIZED  2.0178E-04
>  RMS TSA                              6.7053E-02            NORMALIZED  2.4409E-04
>  RMS TSKIN                            7.0888E-01            NORMALIZED  2.5732E-03
> (npl) ctsm_worktrees/answer_changes> grep RMS /glade/scratch/erik/SMS_Lm1.f19_g17.I1850Clm50BgcCropCmip6waccm.cheyenne_intel.clm-basic.C.20220103_160716_wx0097/run/SMS_Lm1.f19_g17.I1850Clm50BgcCropCmip6waccm.cheyenne_intel.clm-basic.C.20220103_160716_wx0097.clm2.h2.0001-01.nc.cprnc.out
>  RMS EFLX_LH_TOT                      1.1063E+00            NORMALIZED  5.1780E-02
>  RMS FGR                              6.9616E-01            NORMALIZED  7.7931E-02
>  RMS FIRA                             5.9741E-01            NORMALIZED  9.9770E-03
>  RMS FIRE                             5.9741E-01            NORMALIZED  1.8818E-03
>  RMS FSH                              5.9282E+00            NORMALIZED  2.1484E-01
>  RMS FSR                              2.2573E-01            NORMALIZED  2.7303E-03
>  RMS H2OSNO                           8.3972E-01            NORMALIZED  3.5161E-04
>  RMS Q2M                              5.2267E-06            NORMALIZED  1.0254E-03
>  RMS QRUNOFF                          4.8924E-07            NORMALIZED  3.5491E-02
>  RMS RH2M                             2.7443E-01            NORMALIZED  3.3505E-03
>  RMS SNOWDP                           3.2012E-03            NORMALIZED  5.8498E-04
>  RMS TG                               1.2608E-01            NORMALIZED  4.6452E-04
>  RMS TREFMNAV                         6.7270E-02            NORMALIZED  2.5255E-04
>  RMS TREFMXAV                         5.0290E-02            NORMALIZED  1.8243E-04
>  RMS TSA                              5.9663E-02            NORMALIZED  2.2041E-04
>  RMS TSKIN                            1.2627E-01            NORMALIZED  4.6559E-04
>  RMS U10                              1.7895E-02            NORMALIZED  6.2458E-03
>  RMS URBAN_HEAT                       6.0184E-01            NORMALIZED  1.2943E-01
>  RMS WASTEHEAT                        1.1050E-01            NORMALIZED  1.1892E-01
> (npl) ctsm_worktrees/answer_changes> grep RMS /glade/scratch/erik/SMS_Lm1.f19_g17.I1850Clm50BgcCropCmip6waccm.cheyenne_intel.clm-basic.C.20220103_160716_wx0097/run/SMS_Lm1.f19_g17.I1850Clm50BgcCropCmip6waccm.cheyenne_intel.clm-basic.C.20220103_160716_wx0097.clm2.h0.0001-01.nc.cprnc.out
>  RMS CH4PROD                          1.3857E-12            NORMALIZED  4.2508E-05
>  RMS CH4_SURF_AERE_SAT                2.0957E-11            NORMALIZED  9.4182E-04
>  RMS CH4_SURF_DIFF_SAT                3.7061E-13            NORMALIZED  1.7151E-03
>  RMS CH4_SURF_EBUL_SAT                1.4036E-14            NORMALIZED  3.7402E-05
>  RMS EFLXBUILD                        2.0313E-06            NORMALIZED  7.5640E-02
>  RMS EFLX_LH_TOT                      1.4152E-02            NORMALIZED  7.6457E-04
>  RMS ERRH2O                           3.0944E-18            NORMALIZED  5.3811E-03
>  RMS ERRH2OSNO                        4.0004E-20            NORMALIZED  8.4501E-05
>  RMS ERRSEB                           3.8010E-18            NORMALIZED  7.9910E-04
>  RMS ERRSOI                           2.8799E-14            NORMALIZED  8.9140E-05
>  RMS FCH4                             1.6483E-14            NORMALIZED  1.3396E-03
>  RMS FCH4TOCO2                        2.0322E-11            NORMALIZED  7.9741E-04
>  RMS FCH4_DFSAT                       1.8655E-16            NORMALIZED  6.9826E-04
>  RMS FCTR                             9.0260E-03            NORMALIZED  9.4486E-04
>  RMS FGEV                             8.7112E-03            NORMALIZED  1.5175E-03
>  RMS FGR                              5.7613E-03            NORMALIZED  1.0091E-03
>  RMS FGR12                            1.8359E-03            NORMALIZED  4.1239E-04
>  RMS FINUNDATED                       1.5937E-06            NORMALIZED  4.3255E-05
>  RMS FIRA                             7.1961E-03            NORMALIZED  1.2351E-04
>  RMS FIRE                             7.1960E-03            NORMALIZED  2.4181E-05
>  RMS FSA                              1.3684E-03            NORMALIZED  1.6327E-05
>  RMS FSH                              4.9948E-02            NORMALIZED  2.2534E-03
>  RMS FSH_G                            4.9948E-02            NORMALIZED  3.2193E-03
>  RMS FSH_RUNOFF_ICE_TO_LIQ            7.5900E-07            NORMALIZED  1.3786E-05
>  RMS FSH_TO_COUPLER                   4.9948E-02            NORMALIZED  2.2452E-03
>  RMS FSM                              9.5350E-04            NORMALIZED  1.2018E-03
>  RMS FSNO                             1.7435E-04            NORMALIZED  2.8879E-04
>  RMS FSNO_EFF                         1.2966E-04            NORMALIZED  2.1435E-04
>  RMS FSR                              1.3683E-03            NORMALIZED  1.4031E-05
>  RMS FSRND                            3.9501E-04            NORMALIZED  1.1502E-05
>  RMS FSRNDLN                          1.8281E-03            NORMALIZED  2.6464E-05
>  RMS FSRNI                            1.9466E-04            NORMALIZED  2.1726E-05
>  RMS FSRVD                            3.9112E-04            NORMALIZED  1.1327E-05
>  RMS FSRVDLN                          1.8376E-03            NORMALIZED  2.8062E-05
>  RMS FSRVI                            4.1347E-04            NORMALIZED  2.1006E-05
>  RMS FV                               6.5841E-05            NORMALIZED  3.1822E-04
>  RMS H2OSNO                           6.8644E-03            NORMALIZED  2.1790E-06
>  RMS H2OSNO_TOP                       3.0366E-04            NORMALIZED  1.1486E-04
>  RMS HEAT_CONTENT1                    5.1755E+03            NORMALIZED  3.1758E-06
>  RMS HIA                              6.5324E-04            NORMALIZED  3.1390E-05
>  RMS HIA_U                            7.3177E-02            NORMALIZED  4.6373E-03
>  RMS HUMIDEX                          6.8519E-04            NORMALIZED  2.7694E-05
>  RMS HUMIDEX_U                        7.6446E-02            NORMALIZED  4.1531E-03
>  RMS ICE_CONTENT1                     1.2583E-02            NORMALIZED  3.7709E-06
>  RMS LIQUID_CONTENT1                  8.6240E-03            NORMALIZED  5.6716E-06
>  RMS NEM                              2.0345E-11            NORMALIZED  1.8919E-03
>  RMS Q2M                              7.7595E-08            NORMALIZED  1.8397E-05
>  RMS QDRAI                            2.1788E-09            NORMALIZED  3.8732E-04
>  RMS QDRAI_PERCH                      3.4124E-16            NORMALIZED  3.0210E-03
>  RMS QDRAI_XS                         2.5341E-09            NORMALIZED  4.9219E-01
>  RMS QFLX_EVAP_TOT                    5.5856E-09            NORMALIZED  7.5921E-04
>  RMS QFLX_LIQDEW_TO_TOP_LAYER         4.3722E-10            NORMALIZED  3.4617E-03
>  RMS QFLX_LIQEVAP_FROM_TOP_LAYER      2.3623E-09            NORMALIZED  1.2540E-03
>  RMS QFLX_SNOW_DRAIN                  3.0792E-09            NORMALIZED  2.7157E-03
>  RMS QFLX_SOLIDDEW_TO_TOP_LAYER       1.3130E-10            NORMALIZED  4.9122E-04
>  RMS QFLX_SOLIDEVAP_FROM_TOP_LAYER    2.0850E-09            NORMALIZED  6.8169E-03
>  RMS QINFL                            2.0144E-09            NORMALIZED  2.2652E-04
>  RMS QOVER                            2.4735E-09            NORMALIZED  1.2933E-03
>  RMS QRGWL                            6.8235E-12            NORMALIZED  8.4377E-06
>  RMS QRUNOFF                          3.7758E-09            NORMALIZED  4.6135E-04
>  RMS QRUNOFF_TO_COUPLER               3.7758E-09            NORMALIZED  4.6135E-04
>  RMS QSNOEVAP                         3.3256E-09            NORMALIZED  1.4791E-03
>  RMS QSNOFRZ                          2.0552E-09            NORMALIZED  1.0482E-03
>  RMS QSNOMELT                         2.8574E-09            NORMALIZED  1.2018E-03
>  RMS QSOIL                            3.3256E-09            NORMALIZED  1.4771E-03
>  RMS QVEGT                            3.6089E-09            NORMALIZED  9.4486E-04
>  RMS RH2M                             2.5162E-03            NORMALIZED  2.9765E-05
>  RMS SABG                             1.3684E-03            NORMALIZED  2.6696E-05
>  RMS SNOBCMCL                         1.3037E-10            NORMALIZED  7.2816E-05
>  RMS SNOBCMSL                         5.0590E-11            NORMALIZED  1.5142E-03
>  RMS SNODSTMCL                        1.9178E-08            NORMALIZED  4.2788E-05
>  RMS SNODSTMSL                        7.9049E-09            NORMALIZED  3.2038E-04
>  RMS SNOOCMCL                         1.0871E-09            NORMALIZED  9.0889E-05
>  RMS SNOOCMSL                         4.1794E-10            NORMALIZED  1.1340E-03
>  RMS SNOTXMASS                        1.8477E+00            NORMALIZED  2.4396E-06
>  RMS SNOWDP                           2.7023E-05            NORMALIZED  3.7532E-06
>  RMS SNOWICE                          6.7484E-03            NORMALIZED  2.1437E-06
>  RMS SNOWLIQ                          1.7752E-04            NORMALIZED  8.0043E-05
>  RMS SNOW_DEPTH                       2.7023E-05            NORMALIZED  3.7465E-06
>  RMS SNOW_SINKS                       5.6847E-09            NORMALIZED  1.9539E-03
>  RMS SNOW_SOURCES                     6.6484E-09            NORMALIZED  1.0206E-03
>  RMS SWBGT                            3.9568E-04            NORMALIZED  3.0521E-05
>  RMS SWBGT_U                          4.4036E-02            NORMALIZED  2.8535E-03
>  RMS TAUX                             3.8604E-05            NORMALIZED  5.8359E-04
>  RMS TAUY                             3.8604E-05            NORMALIZED  5.8359E-04
>  RMS TBUILD                           1.7809E-02            NORMALIZED  6.1061E-05
>  RMS TG                               1.4545E-03            NORMALIZED  5.4579E-06
>  RMS TH2OSFC                          1.4410E-03            NORMALIZED  5.4032E-06
>  RMS TOTCOLCH4                        9.6064E-08            NORMALIZED  2.2613E-07
>  RMS TREFMNAV                         8.0765E-04            NORMALIZED  3.0863E-06
>  RMS TREFMXAV                         4.7163E-04            NORMALIZED  1.7389E-06
>  RMS TSA                              6.3591E-04            NORMALIZED  2.3890E-06
>  RMS TSKIN                            1.4516E-03            NORMALIZED  5.4526E-06
>  RMS TWS                              7.9516E-03            NORMALIZED  4.3356E-07
>  RMS U10                              2.1184E-04            NORMALIZED  7.4647E-05
>  RMS U10_DUST                         3.6426E-05            NORMALIZED  1.3637E-05
>  RMS URBAN_HEAT                       2.8019E-03            NORMALIZED  5.3723E-02
>  RMS VOLR                             7.6642E+04            NORMALIZED  3.5044E-04
>  RMS VOLRMCH                          4.7034E+04            NORMALIZED  2.7226E-04
>  RMS WASTEHEAT                        5.5954E-04            NORMALIZED  5.3640E-02
>  RMS WBT                              4.4634E-04            NORMALIZED  2.3196E-05
>  RMS WBT_U                            5.6794E-02            NORMALIZED  4.5506E-03

ekluzek commented on src/biogeophys/UrbBuildTempOleson2015Mod.F90:
```diff
@@ -899,6 +900,14 @@ subroutine BuildingTemperature (bounds, num_urbanl, filter_urbanl, num_nolakec,
+         ! of urban area. eflx_urban_ac and eflx_urban_heat are treated similarly below. This flux is balanced
+         ! by an equal and opposite flux into/out of the building and so has a net effect of zero on the energy balance
+         ! of the urban landunit.
+         eflx_ventilation(l) = wtlunit_roof(l) * ( - ht_roof(l)*(vent_ach/3600._r8) &
```
We should use secsphr for 3600._r8 here.

olyson replied:
Agreed.  There are actually four instances of that in that module. It might be easier if you change them if you don't mind.

ekluzek replied:
Yep, I'll do it.

ekluzek replied:
Noted this in #1653

ekluzek commented on src/biogeophys/EnergyFluxType.F90:
```diff
@@ -276,7 +280,7 @@ subroutine InitAllocate(this, bounds)
     
   !------------------------------------------------------------------------
-  subroutine InitHistory(this, bounds, is_simple_buildtemp)
+  subroutine InitHistory(this, bounds, is_simple_buildtemp, is_prog_buildtemp)
```
@olyson this isn't on you (really it's on me), but I'm wondering about sending both is_simple_buildtemp, and is_prog_buildtemp down to InitHistory. Really they exclusive they can't both be true at the same time (or both false at the same time). But, the way they are done you need to use both to really be sure of the behavior. So how you handled it is correct, but it's causing me to rethink how this interface works. I'm not quite sure I see what a better solution would be though. But, I think it's worth thinking about.

olyson replied:
I can see your point.  An alternative is to use "if ( .not. is_simple_buildtemp)" but then it's less obvious what's going on.  Not sure whether it's worth changing the behavior at this point.  Once we drop support for CLM4.5 it won't be used, although not sure when that will be.

ekluzek replied:
Noted this in issue #1653

ekluzek reviewed (changes requested):
This is awesome @olyson. Thanks for putting this together. The code is pretty small, but I'm sure it took a lot of effort to get right.

I have one small change that I request for sure. It's small so I'm happy to do it.

The other thing I bring up is more open ended, because I'm not sure what the right answer is. So I might do something with that as well. And possibly we should chat about it to come up with something that would be more clear.

olyson commented:
I was interested in seeing how the differences you have posted here compare to my expectations.  However, when I try those comparisons I don't get any output (no RMS), e.g.,

grep RMS /glade/scratch/erik/SMS_Lm1.f19_g17.I1850Clm50BgcCropCmip6waccm.cheyenne_intel.clm-basic.C.20220103_160716_wx0097/run/SMS_Lm1.f19_g17.I1850Clm50BgcCropCmip6waccm.cheyenne_intel.clm-basic.C.20220103_160716_wx0097.clm2.h2.0001-01.nc.cprnc.out

Am I doing something wrong or are those the wrong output files?

ekluzek commented:
@olyson, sorry I did a comparison to a different branch in the same case which wiped out the earlier results. I should've created a new case...

ekluzek commented:
In doing standard testing on cheyenne there are a few multi-year tests that are failing exact restart. I've merged in other changes so the culprit could be other things. But, the tests that are failing are:

FAIL ERP_Ly3_P72x2.f10_f10_mg37.IHistClm50BgcCrop.cheyenne_intel.clm-cropMonthOutput COMPARE_base_rest
FAIL ERS_Ly3_P72x2.f10_f10_mg37.IHistClm50BgcCropG.cheyenne_intel.clm-cropMonthOutput COMPARE_base_rest
FAIL ERS_Ly5_P144x1.f10_f10_mg37.IHistClm51BgcCrop.cheyenne_intel.clm-cropMonthOutput COMPARE_base_rest
FAIL ERS_Ly5_P72x1.f10_f10_mg37.IHistClm45BgcCrop.cheyenne_intel.clm-cropMonthOutput COMPARE_base_rest

Multiple fields are different to a large degree in all history fields, so this is an important thing to solve.

olyson commented:
The clm_short suite had passed on my branch.  Let me try those particular tests on my branch and I'll let you know.

olyson commented:
Those four tests pass on my branch, including COMPARE_base_rest, with bfb baseline comparison for the CLM4.5 test as expected.
So it seems like it must be due to something else or an interaction of this with other changes.

ekluzek commented:
> Those four tests pass on my branch, including COMPARE_base_rest, with bfb baseline comparison for the CLM4.5 test as expected.
> So it seems like it must be due to something else or an interaction of this with other changes.

Perfect thanks for checking @olyson that will help me tracking this down. I might just do a git bisect to figure it out.

ekluzek commented:
OK, I find this really odd. The bisect showed me that ctsm5.1.dev073 is the problem. Which is odd because that's the Gregorian calendar fixes, and I don't readily see why that would cause a problem here.

I am able to get the test to pass if I zero out eflx_ventilation in the latest version. So it does seem to be something in the interaction of dev073 with ventilation flux.

olyson commented:
That is pretty odd for sure.  I see that some of the RMS differences are in rural (_R) variables. The ventilation flux should only be affecting urban landunits.  I've been looking at the code but I don't see anything so far...

olyson commented:
I checked out ctsm5.1.dev073 and put in the changes from my original urbanvent branch:

        modified:   src/biogeophys/BalanceCheckMod.F90
        modified:   src/biogeophys/EnergyFluxType.F90
        modified:   src/biogeophys/SoilFluxesMod.F90
        modified:   src/biogeophys/SoilTemperatureMod.F90
        modified:   src/biogeophys/UrbBuildTempOleson2015Mod.F90

and those tests passed.

I then checked out ctsm5.1.dev074  and did the same thing and those tests passed.
Which seems to indicate the urbanvent is not a problem?

ekluzek commented:
OK, I noticed two new bits of information. One is that if I turn transient PFT's off (do_transient_pfts and do_transient_crops to FALSE) the test passes normally. The other is that there are many other restart tests that span more than one year, the difference with these is that they do the restart on the year boundary. So I think a requirement for the problem is to restart on the year boundary and have transient PFT's on. As such I think I can construct a shorter test that shows the problematic behavior running over the year boundary if the restart happens right on the year boundary.

ekluzek commented:
Hey Keith

That can mean there is some interaction with something else, or you did
something subtle in your by hand update that fixed something relative to
the git update I did. Can you point me to your sandbox so I can compare?

Thanks

On Sun, Feb 13, 2022, 1:47 PM Keith Oleson ***@***.***> wrote:

> I checked out ctsm5.1.dev073 and put in the changes from my original
> urbanvent branch:
>
>     modified:   src/biogeophys/BalanceCheckMod.F90
>     modified:   src/biogeophys/EnergyFluxType.F90
>     modified:   src/biogeophys/SoilFluxesMod.F90
>     modified:   src/biogeophys/SoilTemperatureMod.F90
>     modified:   src/biogeophys/UrbBuildTempOleson2015Mod.F90
>
> and those tests passed.
>
> I then checked out ctsm5.1.dev074 and did the same thing and those tests
> passed.
> Which seems to indicate the urbanvent is not a problem?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ESCOMP/CTSM/pull/1535#issuecomment-1038417696>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACYCZQGIEEISOOW6DCA7EDDU3AKEVANCNFSM5G3MGSGQ>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>

ekluzek commented:
OK, the changes for the urbanvent files you have are identical to mine. The two things that differ are the addition of the fire-emission changes and then updated externals. These tests seem to work on the fire-emission branch when taken alone. So possibly it's an interaction of these changes with fire-emission and/or externals change.

Also I tried doing a test where the restart happens at the beginning of the year, and that worked correctly. So I'm not able to reproduce this problem with a shorter test yet.

I tried two tests of this in the fire_emis_fix branch. In the first (that worked) I just did the minimum update in externals that was required. Just now I tried a test (that failed) where I updated to the same externals as are here. So this problem may  just have to do with one of the externals.

ekluzek commented:
It's looking like it's a CDEPS external update. And it looks like it's the update in CDEPS that fixes the solar zenith angle forcing. I'm not sure why that update would screw up some of the restarts. I'm doing more work to make sure I've narrowed it down to the exact tag that causes problems.
